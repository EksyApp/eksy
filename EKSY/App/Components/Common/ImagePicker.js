import React, {Component} from 'react'
import Button from "./Button";
import Picker from 'react-native-image-picker'
import {Image} from 'react-native'
import PropTypes from 'prop-types';

export default class ImagePicker extends Component {
	
	render() {
		return (
				<Button onPress={() => this._buttonClick()}>
					{this.props.buttonText}
				</Button>
		)
	}
	
	_buttonClick() {
		let options = {
			title: 'Select image',
			noData: true,
			storageOptions: {
				skipBackup: true,
			},
			maxWidth: 800,
			maxHeight: 800
		}
		
		options = this.props.options != null ? this.props.options : options
		
		Picker.showImagePicker(options, (response) => {
			
			if (response.didCancel) {
				this.props.onPickerCancelled != null ? this.props.onPickerCancelled() : false;
			} else if (response.error) {
				this.props.onPickerError != null ? this.props.onPickerError(response.error) : false
			} else {
				this._addImage(response.uri)
			}
		})
	}
	
	_addImage(uri) {
		Image.getSize(uri, (width, height) => {
			this._imageUriWorks(uri, width, height)
		}, (error) => {
			this._imageUriError(error)
		})
	}
	
	_imageUriWorks(uri, width, height) {
		this.props.onNewImage({uri, width, height})
	}
	
	_imageUriError(error) {
		this.props.onUriError !== null ? this.props.onUriError(error) : false;
	}
}

ImagePicker.PropTypes = {
	buttonText: PropTypes.string,
	options: PropTypes.object,
	onPickerCancelled: PropTypes.func,
	onPickerError: PropTypes.func,
	onNewImage: PropTypes.func,
	onUriError: PropTypes.func,
}