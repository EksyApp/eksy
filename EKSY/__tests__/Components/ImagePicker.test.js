import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ImagePicker from "../../App/Components/Common/ImagePicker";

test('ImagePicker renders correctly', () => {
	const tree = renderer.create(
			<ImagePicker
					buttonText=""
					onPickerCancelled={() => {}}
					onPickerError= {() => {}}
					onNewImage={() => {}}
					onUriError= {() => {}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});