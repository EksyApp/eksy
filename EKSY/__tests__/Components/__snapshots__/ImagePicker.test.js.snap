// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "rgb(230,232,227)",
      "borderColor": "rgb(101,114,122)",
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 40,
      "marginBottom": 15,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 5,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "rgb(101,114,122)",
        "fontSize": 16,
        "fontWeight": "600",
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    
  </Text>
</View>
`;
